### ISSUE TITLE
Microservices as process vs microservices as containers

### ISSUE THREAD

**Issue by fazilmajeed** [Microsoft/service-fabric#169]
Hi , 
I have a question regarding the microservices as process and microservices as containers..i know sf is capable to accomodate both approach..but which one is most prefferable approach interms of performance?

As we know containers are isolated. And it is not sharing and using resources out the container. Does it cause to performance impact?? And do the process approach sharing the host resources??..

Which is the most prefferable approch if we are targeting only to service fabrics?? No need to port the application in to any other orchastrator..

With Thanks
Fasil Majeed..

**Comment by samedder** [comment]
They're different technologies. Microservices are just a mechanism by which you partition your workload, usually built ontop of a transactional model, or actor model.

Containers are a form of isolation and packaging.


From: Fazil Majeed
Sent: Friday, June 29, 9:48 AM
Subject: [Microsoft/service-fabric] Microservices as process vs microservices as containers (#169)
To: Microsoft/service-fabric
Cc: Subscribed


Hi ,
I have a question regarding the microservices as process and microservices as containers..i know sf is capable to accomodate both approach..but which one is most prefferable approach interms of performance?
As we know containers are isolated. And it is not sharing and using resources out the container. Does it cause to performance impact?? And do the process approach sharing the host resources??..
Which is the most prefferable approch if we are targeting only to service fabrics?? No need to port the application in to any other orchastrator..
With Thanks
Fasil Majeed..

**Comment by fazilmajeed** [comment]
Yes, Microservices are domain level seperation of the application in to small services.. 
As per the servicefabrics application model  we can create an application with multiple services. And can publish it to service fabrics cluster..and there we have the

**Comment by nareshkhatri81** [comment]
With containers, we get portablity of moving from one ochestator like SF to K8s.

With containers, you can still use SF feature of Reverse proxy and naming service to locate services for S2S communication

With performance, I don't think containers will be less performanent.

**Comment by olitomlinson** [comment]
From my understanding, its worth considering **cost** when making decision to use service-per-container (Mesh) vs service-per-process (Native).

For instance, I have a setup for Native where I have around 700 services at any one time (each service is a tenant). On Native I can _pack_ these 700 services across a 5 node cluster (which costs approx £250 a month).

If I was to take this approach into Mesh, I would be using 700 windows container Instances 

700 (instances) * 744 (hours) * 60 (mins) * 60 (seconds) * £0.000011 (1 vCPU = container compute duration + software duration) 
= £20,000

minimum vCPU size is 0.5 so lets halve the cost so far 
= £10,000

This doesn't even take into account the GB-s memory charging, or charging for volume storage or reliable collections.

So, £250 for using Service Fabric Native, or £10,000~ for using Mesh.

How much do you need performance isolation between your services is the question? There is a considerable cost if you do NEED performance isolation.
